ci/test/cargo-test/image/run-tests

```diff
@@ -19,8 +19,7 @@ set -euo pipefail
 ci_init
 
 while read -r binary cwd; do
-    (cd "$cwd" && ci_try "/tests/$binary" --nocapture -Z unstable-options --format json --report-time | tee results.json)
-    cd "$OLDPWD"
+    (cd "$cwd" && ci_try "/tests/$binary" --nocapture -Z unstable-options --format json --report-time | tee results.json && cd "$OLDPWD")
```

Comment:
You can ditch the `cd "$OLDPWD"` because the `cd` now executes in a subshell, so only lasts for the duration of the subshell.